gadget/device_linux.go

```diff
@@ -137,3 +200,45 @@ func FindMountPointForStructure(ps *PositionedStructure) (string, error) {
 
 	return mountPoint, nil
 }
+
+const writableFsLabel = "writable"
+
+func findParentDeviceWithWritableFallback() (string, error) {
+	byWritableLabel := filepath.Join(dirs.GlobalRootDir, "/dev/disk/by-label/", writableFsLabel)
+	target, err := filepath.EvalSymlinks(byWritableLabel)
+
+	if err != nil {
+		if os.IsNotExist(err) {
+			return "", ErrDeviceNotFound
+		}
+		return "", fmt.Errorf("cannot resolve device symlink for filesystem label %q: %v", writableFsLabel, err)
```

Comment:
Do we have a test for this and not-exists error condition above?

Reply:
IsNotExist is already tested, I'll add something for the else case.